lib/jaeger/encoders/thrift_encoder.rb

```diff
@@ -156,10 +156,18 @@ def empty_batch_size_cache
         @empty_batch_size_cache ||= {}
       end
 
-      def empty_batch_size(protocol_class)
+      def caclulate_empty_batch_size(protocol_class)
         empty_batch_size_cache[protocol_class] ||=
           message_size(encode_batch([]), protocol_class) + BATCH_SPANS_SIZE_WINDOW
       end
+
+      def calculate_max_spans_length(protocol_class, max_message_length)
+        empty_batch_size = caclulate_empty_batch_size(protocol_class)
+        max_spans_length = max_message_length - empty_batch_size
+        return max_spans_length if max_spans_length.positive?
+
+        raise "Batch header have size #{empty_batch_size}, but limit #{max_message_length}"
```

Comment:
I initially thought this was going to stop the reporter instance, but I guess it cannot really operate anyway if the header size is larger than the limit. I'm assuming this doesn't interfere with the application itself, so it should be okay change.